WordPress/WordPressHomeWidgetToday/WordPressHomeWidgetToday.swift

```diff
@@ -67,12 +68,102 @@ private extension Provider {
         // refresh interval of the widget, in minutes
         static let refreshInterval = 60
     }
+}*/
+
+struct SiteListProvider: IntentTimelineProvider {
+
+    private var defaultSiteID: Int? {
+        // TODO - TODAYWIDGET: taking the default site id from user defaults for now.
+        // This would change if the old widget gets reconfigured to a different site than the default.
+        // This will be updated with the configuration intent.
+        UserDefaults(suiteName: WPAppGroupName)?.object(forKey: WPStatsTodayWidgetUserDefaultsSiteIdKey) as? Int
+    }
+
+    private func widgetData(for siteID: String) -> HomeWidgetTodayData? {
+        // TODO - TODAYWIDGET: we might change this, but for now an ID equal to zero should not return any valid data
+        //HomeWidgetTodayData.read()?[siteID ?? 0]
+
+        /// - TODO: we should not really be needing to do this conversion.  Maybe we can evaluate a better mechanism for site identification.
+        guard let siteID = Int(siteID) else {
+            return nil
+        }
+
+        return HomeWidgetTodayData.read()?[siteID]
+    }
+
+    // TODO - TODAYWIDGET: This can serve as static content to display in the preview if no data are yet available
+    // we should define what to put in here
+    static let staticContent = HomeWidgetTodayData(siteID: 0,
+                                                   siteName: "Places you should visit",
+                                                   iconURL: nil,
+                                                   url: "",
+                                                   timeZoneName: "GMT",
+                                                   date: Date(),
+                                                   stats: TodayWidgetStats(views: 5980,
+                                                                           visitors: 4208,
+                                                                           likes: 107,
+                                                                           comments: 5))
+
+    // refresh interval of the widget, in minutes
+    static let refreshInterval = 60
+
+    func placeholder(in context: Context) -> HomeWidgetTodayData {
+        Self.staticContent
+    }
+
+    func getSnapshot(for configuration: SelectSiteIntent, in context: Context, completion: @escaping (HomeWidgetTodayData) -> Void) {
+
+        /// - TODO: review this guard... it's crazy we'd have to ever show static content.  Maybe we need to show an error message?
+        ///
+        guard let site = configuration.site,
+              let siteIdentifier = site.identifier,
+              let widgetData = widgetData(for: siteIdentifier) else {
+
+            completion(Self.staticContent)
```

Comment:
I'll leave this for future evaluation, since I'm not sure it's safe NOT to call the completion closure.

Personally I'd expect the `completion` closure to always be called, one ways or another, since it doesn't specifically signal a "success" situation, but the end of an operation.

Reply:
as far as I understood, `getSnapshot` is used in transient situations, like when the user is installing the widget (so it's sort of a preview): in this case I think it's good to show some local data if we have them already. If we don't, we should probably try to understand what kind of information we want to show in that place (the current `staticContent` needs to be changed anyway, or we should use a different view entirely)

Reply:
Marking this as resolved for now as there's no further action required.